### PR TITLE
Create PEM formatted RSA keys for ssh server

### PR DESCRIPTION
**Pull request by waveletlet**
Fix for https://github.com/sosedoff/gitkit/issues/9

### CODE REVIEW

**Comment by waveletlet** [comment]
Replaced call to `ssh-keygen` with key generation using go stdlibs.

File: ssh.go [hunk 1]
```diff
@@ -179,11 +183,33 @@ func (s *SSH) createServerKey() error {
...
+
+	privateKey, err := rsa.GenerateKey(rand.Reader, 2048)
 	if err != nil {
-		return fmt.Errorf("key creating failed: %s", out)
+		return err
 	}
-	return nil
+
+	privateKeyFile, err := os.Create(s.config.KeyPath())
+	fmt.Println(err)
```
Comment (sosedoff) [inline ssh.go]: you can remove this debug statement

File: ssh.go [hunk 2]
```diff
@@ -179,11 +183,33 @@ func (s *SSH) createServerKey() error {
...
-	out, err := exec.Command("ssh-keygen", "-f", s.config.KeyPath(), "-t", "rsa", "-N", "").CombinedOutput()
+
+	privateKey, err := rsa.GenerateKey(rand.Reader, 2048)
 	if err != nil {
-		return fmt.Errorf("key creating failed: %s", out)
+		return err
 	}
-	return nil
+
+	privateKeyFile, err := os.Create(s.config.KeyPath())
```
Comment (sosedoff) [inline ssh.go]: make sure to check for error here before you do chmod

File: ssh.go [hunk 3]
```diff
@@ -179,11 +183,33 @@ func (s *SSH) createServerKey() error {
...
+	if err != nil {
+		return err
+	}
+	privateKeyPEM := &pem.Block{Type: "RSA PRIVATE KEY", Bytes: x509.MarshalPKCS1PrivateKey(privateKey)}
+	if err := pem.Encode(privateKeyFile, privateKeyPEM); err != nil {
+		return err
+	}
+
+	pubKeyPath := s.config.KeyPath() + ".pub"
+	fmt.Println(pubKeyPath)
```
Comment (sosedoff) [inline ssh.go]: remove debug statement

**Comment by sosedoff** [comment]
@waveletlet ping

**Comment by waveletlet** [comment]
Ok, fixed the issues you mentioned.

**Pull request merged**